File: kotlin/src/main/kotlin/com/murex/tbw/client/Order.kt
/*******************************************************************************
 *
 * Copyright (c) {2003-2021} Murex S.A.S. and its affiliates.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the MIT License
 * which accompanies this distribution, and is available at
 * https://opensource.org/licenses/MIT
 *
 *******************************************************************************/

package com.murex.tbw.client

import com.murex.tbw.domain.book.Book
import com.murex.tbw.purchase.Invoice

interface Order {

    fun addBook(book: Book, quantity: Int)

    fun checkOut(): Invoice

    fun getQuantityOf(book: Book): Int

}

File: scala/src/main/scala/com/murex/tbw/client/Order.scala
/*
 * ******************************************************************************
 *  *
 *  * Copyright (c) {2003-2022} Murex S.A.S. and its affiliates.
 *  * All rights reserved. This program and the accompanying materials
 *  * are made available under the terms of the MIT License
 *  * which accompanies this distribution, and is available at
 *  * https://opensource.org/licenses/MIT
 *  *
 *  ******************************************************************************
 */

package com.murex.tbw.client

import com.murex.tbw.domain.book.Book
import com.murex.tbw.purchase.Invoice

trait Order {
  def addBook(book: Book, quantity: Int): Unit
  def checkOut(): Invoice
  def getQuantityOf(book: Book): Int
}
